srml/executive/src/lib.rs

```diff
@@ -184,7 +186,9 @@ impl<
 		// decode parameters and dispatch
 		let (f, s) = xt.deconstruct();
 		let r = f.dispatch(s.into());
-		<system::Module<System>>::note_applied_extrinsic(&r, encoded_len as u32);
+		if !is_inherent {
+			<system::Module<System>>::note_applied_extrinsic(&r, encoded_len as u32);
```

Comment:
This breaks multiple assumptions about the code, the function is also:
1. Increasing the current `extrinsic_index` in storage
2. Calculates total length of extrinsics.

If we really don't want to have the event it should rather be passed as a flag.